app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,7 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    @timeline_posts ||= current_user.friends_post(current_user).ordered_by_most_recent
```

Comment:
Update this. 
Add this into the User model and use it.
```rb

def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```